lib/bloc/user_profile/user_profile_bloc.dart

```diff
@@ -1,33 +1,74 @@
 import 'dart:async';
 import 'dart:io';
 
+import 'package:breez_translations/breez_translations_locales.dart';
+import 'package:c_breez/bloc/user_profile/default_profile_generator.dart';
 import 'package:c_breez/bloc/user_profile/user_profile_state.dart';
 import 'package:c_breez/models/user_profile.dart';
 import 'package:c_breez/services/breez_server.dart';
 import 'package:c_breez/services/notifications.dart';
+import 'package:fimber/fimber.dart';
 import 'package:hydrated_bloc/hydrated_bloc.dart';
 import 'package:path_provider/path_provider.dart';
 
 const PROFILE_DATA_FOLDER_PATH = "profile";
 
+final _log = FimberLog("UserProfileBloc");
+
 class UserProfileBloc extends Cubit<UserProfileState> with HydratedMixin {
   final BreezServer _breezServer;
   final Notifications _notifications;
 
-  UserProfileBloc(this._breezServer, this._notifications)
-      : super(UserProfileState.initial());
+  UserProfileBloc(
+    this._breezServer,
+    this._notifications,
+  ) : super(UserProfileState.initial()) {
+    stream.listen((profile) {
+      _log.v("State: ${profile.profileSettings.toJson()}");
+      final settings = profile.profileSettings;
+      if (settings.color == null || settings.animal == null) {
+        _log.v("Profile has not color or name, generating new random ones…");
+        final defaultProfile = generateDefaultProfile();
+        emit(profile.copyWith(
+          profileSettings: settings.copyWith(
+            color: settings.color ?? defaultProfile.color,
+            animal: settings.animal ?? defaultProfile.animal,
+          ),
+        ));
+      } else if (settings.name == null) {
```

Comment:
why `else if` on first install we won't have color, animal, name and notification registered so we need to fill/generate any of this properties unconditionally.

Reply:
Because it is in the stream, it will be filled in three steps; I broke it into small steps to clarify what is being changed and why; maybe it is too much from my side.

Reply:
Checking all at one single step (kind of, still break each part)